modules/cli/src/main/scala/scala/cli/commands/SetupIde.scala

```diff
@@ -148,14 +149,22 @@ object SetupIde extends ScalaCommand[SetupIdeOptions] {
     val json                      = gson.toJson(details)
     val scalaCliOptionsForBspJson = write(options.shared)
 
+    if (bspJsonDestination.toString.contains(".cache"))
+      value(Left(new FilePermissionsError(
+        """scala-cli can not save bsp file to destination path.
+          |Check if you are owner directory which contains specified input of scala files.
+          |""".stripMargin
+      )))
+
     if (previousCommandName.isEmpty || !bspJsonDestination.toIO.exists()) {
       os.write.over(bspJsonDestination, json.getBytes(charset), createFolders = true)
       os.write.over(
         scalaCliBspJsonDestination,
         scalaCliOptionsForBspJson.getBytes(charset),
         createFolders = true
       )
-      logger.debug(s"Wrote $bspJsonDestination")
+      if (isRunSafe) logger.debug(s"Wrote $bspJsonDestination")
```

Comment:
We shouldn't need this variable (`runSafe`). If you really want to print a message from `setup-ide`, return `bspJsonDestination` from that method, and use it in the `run` method above.